security/oc_csr_internal.h

```diff
@@ -65,10 +86,10 @@ bool oc_sec_csr_verify_signature(mbedtls_x509_csr *csr, int md_flags);
  * @return 0 on success
  * @return -1 on error
  */
-int oc_sec_csr_validate(const unsigned char *csr, size_t csr_len,
-                        mbedtls_pk_type_t pk_type, int md_flags,
-                        oc_string_t *subject_DN, uint8_t *public_key,
-                        size_t public_key_size);
+int oc_sec_csr_validate2(const unsigned char *csr, size_t csr_len,
```

Comment:
Does this deprecate function `oc_sec_csr_validate`?

Reply:
if yes then we need to use `oc_sec_csr_validate_v1` because we don't want to introduce a new convention.

Reply:
You're too quick! I've pushed this just to get analysis from Sonar, this function will be removed once I rewrite the last test.